### DISCUSSION TITLE
Change getPage() to be StatelessWidget

### DISCUSSION THREAD

**Discussion by hoffmannkrzysztof** [EXERLOG/exer_log#155] [General]
While checking the code. I noticed that there is a function that returns Widget. It should not be like that. If you do, it is a performance issue as this widget cannot be reused in the widget tree and it is rebuilt every time. Its better to replace getPage() to StatelessWidget with two parameters: bool firstLoad, WorkoutData workoutData and use ThemeProvider internally.

**Comment by jorre127** [comment]
Agree with this, I avoid using functions that return widgets as much as possible, definitely something to keep in mind when we rewrite this page.
